### ISSUE TITLE
More dimensions ?

### ISSUE THREAD

**Issue by gabriel-hurtado** [haowen-xu/donut#3]
Hello,

Would it be complicated to add the possibility of using Donut's abilities in a higher dimensional space ?
In theory I do not see a strong reason why, but I wondered if you think that it is feasible, and if so, if you have an idea how.

**Comment by haowen-xu** [comment]
Yes, in theory, it's possible; and in practice, it's easy.  Just change `x_dims` argument, and you can fit Donut to higher dimensional sliding windows.  However, I haven't carried out such experiments.  Also, as the dimension goes higher, the number of trainable parameters goes higher (notice the 100 to x_dims transformation), which might introduce optimization issues. 

I don't see many reasons to extend the window size to a much larger number, where two hours is enough for the Donut situations.  If you want to apply anomaly detection on other types of data, you may need to investigate the optimal solution by yourself.  Sometimes the reconstruction probability may even not applicable.

**Comment by gabriel-hurtado** [comment]
Thank you very much for the answer, and in general for the project, it is really interesting !

However, I think my question was not really clear. 

What I would like to achieve would be to have a reconstruction probability as an output, given n different kpis. I am not interested in changing the size of the window of focus.

**Comment by haowen-xu** [comment]
@gabriel-hurtado Okay, you've got a point.  This is a research topic of another member in my laboratory, but the work is still in progress.

**Comment by gabriel-hurtado** [comment]
Okay, thank you for the answer ! I am greatly interested by this topic, feel free to let me know if I can contribute in any way.
